app/Policies/AlbumPolicy.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Policies;
+
+use App\Models\Album;
+use App\Models\User;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class AlbumPolicy
+{
+	use HandlesAuthorization;
+
+	/**
+	 * Perform pre-authorization checks.
+	 *
+	 * @param \App\Models\User $user
+	 * @param string           $ability
+	 *
+	 * @return void|bool
```

Comment:
agreed. I really don't understand why they want a bool|void return, I don't see the rationale.
I assume it is for a three-state:
- `true` => immediately authorize
- `false` => immediately reject
- `void` => do nothing

Reply:
Yes, it is for a tri-state, but I would have prefered `bool|null`. This would not only allow to use the null coalesce operator `??` instead of a `isset`, but it also avoid retuning nothing by accident. If there is a `return;` one never knows if it was intentionally or if the return value is simply missing. `return null;` would have been more explicit. They actually fall into that trap in `Model::save()` which contains a spurious `return` but should have been a `return true`.

Anyway not our problem.